dataprofiler/profilers/profile_builder.py

```diff
@@ -619,11 +680,8 @@ def _update_row_statistics(self, data, sample_ids=None):
             if null_type_dict:
                 null_row_indices = set.union(*null_type_dict.values())
 
-            if sample_ids is not None:
-                # If sample ids provided, only consider nulls in rows that
-                # were fully sampled
-                null_row_indices = null_row_indices.intersection(
-                    data.index[sample_ids[:self._min_col_samples_used]])
+            null_row_indices = null_row_indices.intersection(
+                data.index[sample_ids[:self._min_col_samples_used]])
```

Comment:
i'm realizing now, that what we could do instead to not lose this speed up of the if statement, is make the add at the bottom be conditional as well.
```python
if sample_ids is not None:
    self.row_has_null_count += len(null_in_row_count)
    self.row_is_null_count += len(null_rows)
else:
    self.row_has_null_count = len(null_in_row_count)
    self.row_is_null_count = len(null_rows)
```